### ISSUE TITLE
abseil package has wrong order of libs

### ISSUE THREAD

**Issue by fulara** [bincrafters/community#393]
I finally got to prepare a test sample proving the pr i raised there: https://github.com/bincrafters/conan-abseil/pull/2 which was subsequently reverted, as it was breaking other things.

I dont think that the current order of libraries in bincrafter's abseil package is the correct one.

### Package Details (Include if Applicable)
* abseil/20180223@bincrafters/stable
* CentOS Linux release 7.5.1804 (Core) 
* gcc 4.8.5

### Steps to reproduce (Include if Applicable)
to reproduce:
`conanfile.py`
```
from conans import ConanFile, CMake, tools
class AbslTestConan(ConanFile):
    name = "absl-test"
    version = "1.0.0"
    settings = "os", "compiler", "build_type", "arch"
    exports_sources = "src/*", "CMakeLists.txt"
    requires = "abseil/20180223@bincrafters/stable"
    generators = "cmake"

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()
```
`cmake:`
```

project (absl-test)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup()

add_executable(absl-test src/main.cpp)
target_link_libraries(absl-test ${CONAN_LIBS})
```
`main.cpp:`
```
#include "absl/strings/str_join.h"

int main() {
	std::string f;
	absl::StrAppend(&f, 5);
	return 5;
}
```
building the project results in:
```
[100%] Linking CXX executable bin/absl-test
/home/fulara/.conan/data/abseil/20180223/bincrafters/stable/package/1cc7e1b3fae71d28a883d673b4e71cee5f153d9f/lib/libabsl_strings.a(numbers.cc.o): In function `absl::SimpleAtob(absl::string_view, bool*)':
numbers.cc:(.text+0x594): undefined reference to `absl::raw_logging_internal::RawLog(absl::LogSeverity, char const*, int, char const*, ...)'
collect2: error: ld returned 1 exit status
```
Hacking recipe moving absl_base to the end works for me, however breaks other things, so it would be nice to know what really should be the order of the libs.

**Comment by madebr** [comment]
Looks like a cyclic dependency. Some libraries should be listed twice (or more).
Or use http://brainychen72.blogspot.com/2017/12/gcc-start-group-and-end-group-command.html,
but I don't think conan supports this and this will not work on msvc.

**Comment by SSE4** [comment]
I can't find pkg-config files for abseil or something like that which specifies correct linking order

**Comment by SSE4** [comment]
might be helpful https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=abseil-cpp-git

**Comment by uilianries** [comment]
Hi! I fixed the library order and released a new version: 20180600

Please, check if works for you.

**Comment by SSE4** [comment]
I am closing this one for now
@fulara if issue persists for you, please re-open
